### ISSUE TITLE
Cannot read property 'toString' of undefined

### ISSUE THREAD

**Issue by AuroraLantean** [Henry-E/dog-money#2]
Hi Henry. Somehow I got this error: 
     TypeError: Cannot read property 'toString' of undefined
      at /mnt/sda5/0Programming/Solana/anchor1/node_modules/@solana/web3.js/lib/index.cjs.js:808:47
      at Array.forEach (<anonymous>)
      at Transaction.compileMessage (/mnt/sda5/0Programming/Solana/anchor1/node_modules/@solana/web3.js/lib/index.cjs.js:807:18)
      at Transaction._compile (/mnt/sda5/0Programming/Solana/anchor1/node_modules/@solana/web3.js/lib/index.cjs.js:912:26)
      at Transaction.partialSign (/mnt/sda5/0Programming/Solana/anchor1/node_modules/@solana/web3.js/lib/index.cjs.js:1044:26)
      at NodeWallet.signTransaction (/mnt/sda5/0Programming/Solana/anchor1/node_modules/@project-serum/anchor/dist/cjs/provider.js:161:12)
      at Provider.send (/mnt/sda5/0Programming/Solana/anchor1/node_modules/@project-serum/anchor/dist/cjs/provider.js:77:27)
      at processTicksAndRejections (node:internal/process/task_queues:94:5)
      at async createMint (tests/utils/index.js:39:3)
      at async Context.<anonymous> (tests/dog-money.js:27:22)

tests/dog-money.js:27:22
`    const usdcMint = await createMint(program.provider, program.provider.wallet.publicKey);`

tests/utils/index.js:39:3)
```
  const tx = new anchor.web3.Transaction();
  tx.add(...instructions);
  console.log("mint:", mint, ", tx:", tx);
  await provider.send(tx, [mint]);
```
console outputs:
mint: Keypair {
  _keypair: {
    publicKey: Uint8Array(32) [171,  95,  92, 181,...  ],
    secretKey: Uint8Array(64) [
       19, 183,  50, 155, ...    ]
  }
}
tx: Transaction {
  signatures: [],
  feePayer: undefined,
  instructions: [
    TransactionInstruction {
      keys: [Array],
      programId: [PublicKey],
      data: <Buffer 00 00 00 00 60 4d 16 00 00 00 00 00 52 00 00 00 00 00 00 00 06 dd f6 e1 d7 65 a1 93 d9 cb e1 46 ce eb 79 ac 1c b4 85 ed 5f 5b 37 91 3a 8c f5 85 7e ff ... 2 more bytes>
    },
    TransactionInstruction {
      keys: [Array],
      programId: [PublicKey],
      data: <Buffer 00 06 79 5f c7 29 5d 79 b5 bd 54 b3 48 36 b7 43 2c ec 9d 1f b4 20 d3 20 bf 0e 13 e1 14 fa 29 b8 b9 0e 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ... 17 more bytes>
    }
  ],
  recentBlockhash: undefined,
  nonceInfo: undefined
}
So somehow this tx object has properties of undefined... how did this happen? 
May I know what went wrong? Thanks

**Comment by Henry-E** [comment]
Thank you for providing so many details but it's still kind of hard to parse what's going on. 

`TypeError: Cannot read property 'toString' of undefined`
This error usually means you're trying to access the public key of an undefined variable. You'll need to work backwards to figure out which variable is undefined and why it hasn't been defined. For example it's possible that you have defined `program.provider.wallet` somehow, though anchor should do that for you automatically 🤷‍♂️ .

**Comment by AuroraLantean** [comment]
I think this issue has disappeared. Thanks
